runtime/timelib_wrapper.cpp

```diff
@@ -167,17 +170,36 @@ bool php_timelib_is_valid_date(int64_t m, int64_t d, int64_t y) {
   return y >= 1 && y <= 32767 && timelib_valid_date(y, m, d);
 }
 
+static timelib_time *timelib_strtotime_leak_safe(const string &time, timelib_error_container **errors) {
+  timelib_time *t = timelib_strtotime(time.c_str(), time.size(), errors, timelib_builtin_db(), timelib_parse_tzfile);
+  // sometimes 't->tz_info' contains allocated timezone object, but sometimes doesn't. And we are unable to merely delete this object
+  // because it can be shared among different e.g. DateTimeImmutable vs DateTime objects. Given that it's quite rare situation this leak is just suppressed.
+  // (it's not a leak in 'php-src' because all 'tz_info *' there are cached. We are unable cache it in kphp because of our memory managment)
```

Comment:
Why are we unable to cache it in kphp? Timelib caches it in heap, doesn't it?

Reply:
Because these objects allocated in script memory, thus its lifetime limited by lifetime of request, but cache leaves forever. php-src cache it, while timelib itself doesn't